app_dart/lib/src/request_handlers/scheduler/vacuum_stale_tasks.dart

```diff
@@ -54,18 +53,20 @@ class VacuumStaleTasks extends RequestHandler<Body> {
   Future<void> _vacuumRepository(gh.RepositorySlug slug) async {
     final DatastoreService datastore = datastoreProvider(config.db);
 
-    final List<FullTask> tasks = await datastore.queryRecentTasks(slug: slug).toList();
+    final FirestoreService firestoreService = await config.createFirestoreService();
+
+    final List<FullTask> tasks = await firestoreService.queryRecentTasks(slug: slug);
     final List<Task> tasksToBeReset = <Task>[];
     for (FullTask fullTask in tasks) {
       final Task task = fullTask.task;
       if (task.status == Task.statusInProgress && task.buildNumber == null) {
-        task.status = Task.statusNew;
-        task.createTimestamp = 0;
+        task.setStatus(Task.statusNew);
+        task.setCreateTimestamp(0);
         tasksToBeReset.add(task);
       }
     }
     log.info('Vacuuming stale tasks: $tasksToBeReset');
-    await datastore.insert(tasksToBeReset);
+    // await datastore.insert(tasksToBeReset);
```

Comment:
We should keep this logic for now before we validated firestore logics work as expected. Existing dashboard (frontend) is still pulling data from Datastore.

You can dup the logics for both Datastore and Firestore.

Reply:
Actually we just want to keep both `Write` logics. Namely, read based on firestore data, and then update both datastore and firestore. We can have a similar function `documentToTask` in datastore task.dart, as `taskToDocument` in firestore task.dart.